analysis/analysis.go

```diff
@@ -207,20 +206,86 @@ func generateMetricTasks(run *v1alpha1.AnalysisRun) []metricTask {
 	return tasks
 }
 
+// resolveArgs resolves args for metricTasks, including secret references
+// returns resolved metricTasks and secrets for log redaction
+func (c *AnalysisController) resolveArgs(tasks []metricTask, args []v1alpha1.Argument, namespace string) ([]metricTask, []string, error) {
+	//create set of secrets for redaction
+	secretSet := map[string]bool{}
+	for i, arg := range args {
+		//if secret specified in valueFrom, replace value with secret value
+		//error if arg has both value and valueFrom
+		if arg.ValueFrom != nil {
+			if arg.Value != nil {
+				err := fmt.Errorf("arg '%s' has both Value and ValueFrom fields", arg.Name)
+				return nil, nil, err
+			}
+			if arg.ValueFrom.SecretKeyRef == nil {
+				err := fmt.Errorf("arg '%s' does not contain a secret reference", arg.Name)
+				return nil, nil, err
+			}
+			name := arg.ValueFrom.SecretKeyRef.Name
+			if name == "" {
+				err := fmt.Errorf("secret referenced by arg %s has ", arg.Name) //TODO: fix message
+				return nil, nil, err
+			}
```

Comment:
With API validation, a user can't submit a rollout with a SecretKeyRef that has an empty name. As a result, we do not need to check that the name is empty.